packages/datadog-plugin-grpc/src/server.js

```diff
@@ -0,0 +1,144 @@
+'use strict'
+
+const Tags = require('../../../ext/tags')
+const TEXT_MAP = require('../../../ext/formats').TEXT_MAP
+const kinds = require('./kinds')
+const { addMethodTags, addMetadataTags, getFilter } = require('./util')
+
+function handleError (span, err) {
+  span.setTag('error', err)
+}
+
+function createWrapHandler (grpc, tracer, config, handler) {
+  const filter = getFilter(config, 'metadata')
+
+  return function wrapHandler (func) {
+    return function funcWithTrace (call, callback) {
+      const metadata = call.metadata
+      const request = call.request
+      const type = this.type
+      const isStream = type !== 'unary'
+      const scope = tracer.scope()
+      const childOf = tracer.extract(TEXT_MAP, metadata.getMap())
+      const span = tracer.startSpan('grpc.request', {
+        childOf,
+        tags: {
+          [Tags.SPAN_KIND]: 'server',
+          'resource.name': handler,
+          'service.name': config.service || `${tracer._service}`,
+          'component': 'grpc'
+        }
+      })
+
+      addMethodTags(span, handler, kinds[type])
+
+      if (request && metadata) {
+        addMetadataTags(span, metadata, filter, 'request')
+      }
+
+      scope.bind(call)
+
+      // Finish the span if the call was cancelled.
+      call.on('cancelled', () => {
+        span.setTag('grpc.status.code', grpc.status.CANCELLED)
```

Comment:
A small style change, but do you want to make `grpc.status.code` into a constant.